File: backend/ShiftMate.API/Models/Availability.cs
namespace ShiftMate.API.Models;

public class Availability
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public DayOfWeek DayOfWeek { get; set; }

    public TimeSpan StartTime { get; set; }

    public TimeSpan EndTime { get; set; }

    public bool IsAvailable { get; set; }

    public Employee Employee { get; set; } = null!;
}

File: frontend/src/types/availability.ts
export interface Availability {
    id: number
    employeeId: number
    employeeName: string
    dayOfWeek: number
    startTime: string
    endTime: string
    isAvailable: boolean
}

export interface CreateAvailabilityRequest {
    employeeId: number
    dayOfWeek: number
    startTime: string
    endTime: string
    isAvailable: boolean
}